plugins/tiles/src/main/java/org/apache/struts2/tiles/StrutsTilesInitializer.java

```diff
@@ -30,12 +30,22 @@
 
 public class StrutsTilesInitializer extends AbstractTilesInitializer {
 
+    public static final String STRUTS_TILES_ALTERNATE_FILERESOURCEHANDLING = "struts.tiles.alternatefileresourcehandling";
+
     private static final Logger LOG = LogManager.getLogger(StrutsTilesInitializer.class);
 
     @Override
     protected ApplicationContext createTilesApplicationContext(ApplicationContext preliminaryContext) {
         ServletContext servletContext = (ServletContext) preliminaryContext.getContext();
 
+        // Opt-in alternate file resource handling for WW-5011
```

Comment:
Sorry but it doesn't look good to me at all! It's not user's business to switch Struts behavior to fix Struts issue! I would like to **add** such urls support to `StrutsApplicationResource` class privately - In this case, the fix should be transparent for outside of that class.

Reply:
Hello @yasserzamani .

_I think I understand your concern_ here, but _changing the existing default behaviour could inadvertently break existing usage_.  For that reason it seemed an opt-in (via context param in this case) would be reasonable (not all that different from use cases where Struts configuration parameters are available to users to modify behaviour).

It would certainly be nicer/cleaner to change `StrutsApplicationResource` transparently, but due to the risk of breaking something, maybe that could be done in Struts 2.6.x instead (where some breaking behaviour can be expected) ?

Reply:
As far as I could understand the issue, I think the issue is `StrutsApplicationResource` cannot handle resources whose have `#` in their url, right? if so, that's why I bold "add" in my above comment :) I meant, now it should **also** support such urls, it doesn't break previous behavior but is union of the previous behavior and newly added one. Am I wrong? Or maybe you mean it isn't doable in any way without breaking backward-compatibility? if so why? it looks strange to me ! e.g. as a quick dirty solution I can think about this:

```java
String myGetPath(url) {
result=url.getPath();
nsIndex=url.indexOf('#');
if(nsIndex>0)
result += url.subString(nsIndex);
return result;
}
```